proposals/stdlib/hex-format.md

```diff
@@ -0,0 +1,320 @@
+            var suffix: String = ""  
+            var removeLeadingZeros: Boolean = false  
+        }
+    }
+}
+```
+
+`BytesHexFormat` and `NumberHexFormat` classes hold format options for `ByteArray` and numeric values, correspondingly.
+`uppercase` option, which common to both `ByteArray` and numeric values, is stored in `HexFormat`.
+
+It's not possible to instantiate a `HexFormat` or its builder directly. The following function is provided instead:
+```
+public inline fun HexFormat(builderAction: HexFormat.Builder.() -> Unit): HexFormat
+```
+
+### Formatting
+
+For formatting, the following extension functions are proposed:
+```
+// Formats the byte array using HexFormat.uppercase and HexFormat.bytes
+public fun ByteArray.toHexString(format: HexFormat = HexFormat.Default): String
+
+public fun ByteArray.toHexString(  
+    startIndex: Int = 0,  
+    endIndex: Int = size,  
+    format: HexFormat = HexFormat.Default  
+): String
+
+// Formats the numeric value using HexFormat.uppercase and HexFormat.number
+// N is Byte, Short, Int, Long, and their unsigned counterparts
+public fun N.toHexString(format: HexFormat = HexFormat.Default): String
+```
+
+### Parsing
+
+It is critical to be able to parse the results of the formatting functions above.
+For parsing, the following extension functions are proposed:
+```
+// Parses a byte array
+public fun String.hexToByteArray(format: HexFormat = HexFormat.Default): ByteArray
+
+// Parses a numeric value
+// N is Byte, Short, Int, Long, and their unsigned counterparts
+public fun String.hexToN(format: HexFormat = HexFormat.Default): String
+```
+
+## Contracts
+
+* When formatting a `ByteArray`, the LF character is used to separate lines.
+* When parsing a `ByteArray`, any of the char sequences CRLF (`"\r\n"`), LF (`"\n"`) and CR (`"\r"`) is considered a valid line separator.
+* Parsing is performed in case-insensitive manner.
+* `NumberHexFormat.removeLeadingZeros` is ignored when parsing.
+* Assigning a non-positive value to `BytesHexFormat.Builder.bytesPerLine/bytesPerGroup` is prohibited.
+  In this case `IllegalArgumentException` is thrown.
+* Assigning a string containing LF or CR character to `BytesHexFormat.Builder.byteSeparator/bytePrefix/byteSuffix`
+  and  `NumberHexFormat.Builder.prefix/suffix` is prohibited. In this case `IllegalArgumentException` is thrown.
+
+### Examples
+
+```
+// Parsing an Int
+"3A".hexToInt() // 58
+// Formatting an Int
+93.toHexString() // "0000005d"
+
+// Parsing a ByteArray
+val macAddress = "001b638445e6".hexToByteArray()
+
+// Formatting a ByteArray
+macAddress.toHexString(HexFormat { bytes.byteSeparator = ":" }) // "00:1b:63:84:45:e6"
+
+// Defining a format and assigning it to a variable
+val threeGroupFormat = HexFormat { upperCase = true; bytes.bytesPerGroup = 2; bytes.groupSeparator = "." }
+// Formatting a ByteArray using a previously defined format
+macAddress.toHexString(threeGroupFormat) // "001B.6384.45E6"
+```
+
+## Alternatives
```

Comment:
Another option is a third-party implementation. Logging, for example, is only available this way.

Reply:
Could you also please provide a link to the library you are referring to?

Reply:
https://github.com/oshai/kotlin-logging